flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerRunner.java

```diff
@@ -55,6 +55,7 @@ public ResourceManagerRunner(
 		final ResourceManagerConfiguration resourceManagerConfiguration = ResourceManagerConfiguration.fromConfiguration(configuration);
 		final SlotManagerFactory slotManagerFactory = new DefaultSlotManager.Factory();
 		final JobLeaderIdService jobLeaderIdService = new JobLeaderIdService(highAvailabilityServices);
+		final HeartbeatService heartbeatService = new HeartbeatService(resourceManagerConfiguration, rpcService.getExecutor());
```

Comment:
Things which are started outside of the `ResourceManager` should also be closed outside of it.